server/auth/auth.service.ts

```diff
@@ -1,35 +1,44 @@
 import * as bcrypt from 'bcrypt';
 import * as jwt from 'jsonwebtoken';
-import { UserDto, UpdateUserCommand } from '../users/users.dto';
+import { UpdateUserCommand } from '../users/users.dto';
 import HttpException from '../core/exceptions/http.exception';
 import { DataStoredInToken, TokenData } from '../auth/auth.interface';
 import { isEmptyObject } from '../core/util';
 import UserService from '../users/users.service';
-import { Users } from '../../database/models/users';
 import { getAppConfig } from '../core/app.config'
 import { RegisterUserCommand, LoginUserCommand } from './auth.dto';
 import { ensureInputIsEmail } from '../core/validation';
+import { UserModel } from '../users/users.model';
 import { recordAnalyticsEvent } from '../core/analytics-event.service';
  
 class AuthService {
 	private usersService = new UserService()
 
-	public async signup(dto: RegisterUserCommand): Promise<UserDto> {
+	public async signup(dto: RegisterUserCommand): Promise<UserModel> {
 		const created = await this.usersService.createUser(dto)
 
-		recordAnalyticsEvent("UserCreated", created.id)
+		recordAnalyticsEvent("UserCreated", created.id!)
 
 		return created
 	}
 
-	public async login(dto: LoginUserCommand): Promise<{ cookie: string, user: UserDto }> {
+	public async login(dto: LoginUserCommand): Promise<{ cookie: string, user: UserModel }> {
 		if (isEmptyObject(dto)) {
 			throw new HttpException(400, "Incorrect input data")
 		}
 
-		ensureInputIsEmail(dto.email)
+		const user = await this.authenticate(dto);
 
-		const user = await Users.findOne({ where: { email: dto.email } })
+		const tokenData = this.createToken(user.id!);
+		const cookie = this.createCookie(tokenData);
+
+		return { cookie, user };
+	}
+
+	public async authenticate(dto: LoginUserCommand) {
+		ensureInputIsEmail(dto.email);
+
+		const user = await this.usersService.getByEmail(dto.email);
```

Comment:
Let's use `Users.findOne` directly in this service and separate it completely from `UserService`. 

We could then speed up the app by optimizing the query here to only return those few fields that we need, without joining with additional tables that we have do in `UserService`.

Reply:
resolved in https://github.com/mbrizic/ratethegame/pull/28/commits/b158c0718600a8adeb6bacdcd5d09c7e8ab913e4